src/foram3d/calculators/faces_processor.ts

```diff
@@ -0,0 +1,52 @@
+/// <reference path="../foram.ts" />
+/// <reference path="../helpers/face.ts" />
+
+module Foram3D.Calculators {
+  export class FacesProcessor {
+    foram: Foram;
+
+    constructor(foram: Foram) {
+      this.foram = foram;
+    }
+
+    sumFaces(magnitude: (face: Helpers.Face) => number): number {
+      var chambers, chamber, otherChamber,
```

Comment:
Javascript, wewnętrznie, robi mniej więcej coś takiego jak tu zrobiłeś, tj. wszystkie definicje zmiennych w scope wynosi na początek. Co daje ciekawe efekty:
```
var fun = function() {
  im_not_defined = "But i work :)";

  console.log(im_not_defined);
  var im_not_defined = "Whaaat?";
  console.log(im_not_defined);
}
```
Powyższy kod nie ma żadnych błędów. Oznacza to też, że każda zmienna w scope która ma tę samą nazwę prowadzi do dokładnie tego samego wycinka pamięci:
```
var fun = function() {
  var i = 0;
  for (var c of [0, 1, 2, 3, 4]) {
    var i = c;
    console.log(i);
  }
  console.log(i);
}
```
`i == 4 => true` na samym końcu wykonania tej funkcji.

Pomędrkowałem, jaki z tego wniosek? Nie trzeba na początku definiować zmiennych przez var, ponieważ nie ma to żadnego wpływu na wydajność. Moim zdaniem jest niepotrzebne, bo zajmuje tylko miejsce i nie informuje o niczym pożytecznym :] `var` może spokojnie pojawiać się tam, gdzie zmienna jest po raz pierwszy używana.